Source/Core/Core/Config/DefaultLocale.cpp

```diff
@@ -0,0 +1,201 @@
+{
+#ifdef _WIN32
+  // Windows codepath: Check the regional information.
+  // More likely to match the user's physical location than locales are.
+  // TODO: Can we use GetUserDefaultGeoName? (It was added in a Windows 10 update)
+  GEOID geo = GetUserGeoID(GEOCLASS_NATION);
+  const int buffer_size = GetGeoInfoW(geo, GEO_ISO2, nullptr, 0, 0);
+  std::vector<wchar_t> buffer(buffer_size);
+  const int result = GetGeoInfoW(geo, GEO_ISO2, buffer.data(), buffer_size, 0);
+  if (result)
+    return TStrToUTF8(buffer.data());
+#endif
+
+  // Generic codepath: Check the locales.
+  // Might be entirely unrelated to someone's actual country (for instance someone in a
+  // non-English-speaking country using en-US), but could also be a perfect match.
+  for (const std::string& locale : GetPreferredLocales())
+  {
+    if (const std::optional<std::string> country_code = TryParseCountryCode(locale))
+      return *country_code;
+  }
+
+  return "";
+}
+
+static std::optional<u8> ComputeDefaultCountry()
+{
+  // clang-format off
+  static constexpr std::array<std::string_view, 178> COUNTRIES = {
+    "--", "JP", "--", "--", "--", "--", "--", "--", "AI", "AG", "AR", "AW", "BS", "BB", "BZ", "BO",
+    "BR", "VG", "CA", "KY", "CL", "CO", "CR", "DM", "DO", "EC", "SV", "GF", "GD", "GP", "GT", "GY",
+    "HT", "HN", "JM", "MQ", "MX", "MS", "AN", "NI", "PA", "PY", "PE", "KN", "LC", "VC", "SR", "TT",
+    "TC", "US", "UY", "VI", "VE", "--", "--", "--", "--", "--", "--", "--", "--", "--", "--", "--",
+    "AL", "AU", "AT", "BE", "BA", "BW", "BG", "HR", "CY", "CZ", "DK", "EE", "FI", "FR", "DE", "GR",
+    "HU", "IS", "IE", "IT", "LV", "LS", "LI", "LT", "LU", "MK", "MT", "ME", "MZ", "NA", "NL", "NZ",
+    "NO", "PL", "PT", "RO", "RU", "RS", "SK", "SI", "ZA", "ES", "SZ", "SE", "CH", "TR", "GB", "ZM",
+    "ZW", "AZ", "MR", "ML", "NE", "TD", "SD", "ER", "DJ", "SO", "--", "--", "--", "--", "--", "--",
+    "TW", "--", "--", "--", "--", "--", "--", "--", "KR", "--", "--", "--", "--", "--", "--", "--",
+    "HK", "MO", "--", "--", "--", "--", "--", "--", "ID", "SG", "TH", "PH", "MY", "--", "--", "--",
+    "CN", "--", "--", "--", "--", "--", "--", "--", "AE", "IN", "EG", "OM", "QA", "KW", "SA", "SY",
+    "BH", "JO",
+  };
+  // clang-format on
+
+  std::string country = ComputeDefaultCountryCode();
+
+  // Netherlands Antilles was split into three new codes after the release of the Wii
+  if (country == "BQ" || country == "CW" || country == "SX")
+    country = "AN";
+
+  const auto it = std::find(COUNTRIES.cbegin(), COUNTRIES.cend(), country);
+  if (it == COUNTRIES.cend())
+    return std::nullopt;
+
+  return static_cast<u8>(it - COUNTRIES.cbegin());
+}
+
+static DiscIO::Region ComputeDefaultRegion()
+{
+  const std::optional<u8> country = GetDefaultCountry();
+
+  if (country)
+  {
+    const DiscIO::Region region = DiscIO::SysConfCountryToRegion(GetDefaultCountry());
+    ASSERT(region != DiscIO::Region::Unknown);
+    return region;
+  }
+  else
```

Comment:
Unless the policy has changed recently, I think we typically omit else after a return